libticonv/trunk/src/ticonv.h

```diff
@@ -202,10 +202,10 @@ typedef struct
 	/* Convert at most *INBYTESLEFT bytes from *INBUF according to the
 	   code conversion algorithm specified by CD and place up to
 	   *OUTBYTESLEFT bytes in buffer at *OUTBUF.  */
-	TIEXPORT4 size_t TICALL ticonv_iconv (ticonv_iconv_t cd, char ** restrict inbuf,
-	                                      size_t * restrict inbytesleft,
-	                                      char ** restrict outbuf,
-	                                      size_t * restrict outbytesleft);
+	TIEXPORT4 size_t TICALL ticonv_iconv (ticonv_iconv_t cd, char ** __restrict inbuf,
```

Comment:
Silly idea: what about
```
#ifdef __cplusplus
#define restrict __restrict
#endif
```
earlier in that file, possibly fused with the `extern "C" {` block while at it ?
I mean, it would be a real disgrace not to be able to make portable use of a C language keyword when compiling as C...

Reply:
I'll try more things and let you know.

Reply:
Looks like I have found a workaround that works...

Reply:
Heh, macro replacement with a temporary...